src/micro.js

```diff
@@ -0,0 +1,32 @@
+// c is the current script element
+var c = document.currentScript,
+  // d is the dataset of the current script element
+  d = c.dataset,
+  // q is the URL search params
+  q = new URLSearchParams(location.search),
+  // v is the rate of the current script element
+  v = q.get('rum') || q.get('optel') || d.rate,
+  // w is the rate of the current script element
+  w = { on: 1, off: 0, high: 10, low: 1e3 }[v] || 100,
+  // r is the rum object, we initialize it if it doesn't exist
+  r = (window.hlx = window.hlx || {}).rum ||
+    // generate a random id for the rum object
+    (window.hlx.rum = { id: crypto.randomUUID().slice(-9) });
```

Comment:
```suggestion
  h = window.hlx || {};
  r = r.rum || (h.rum = { id: crypto.randomUUID().slice(-9) });
    // generate a random id for the rum object
```

Reply:
```suggestion
  // remember hlx
  h = window.hlx || {},
  // generate a random id for the rum object
  r = h.rum || (h.rum = { id: crypto.randomUUID().slice(-9) });
```

@kptdobe don't you mean this?
- comma instead of semicolon, so we stay in the `var` block
- `h.rum` instead of `r.rum` as `r` is only being defined here

Reply:
I think both suggestions are wrong, the initial code is setting window.hlx if it is not set
Both suggestions are not setting it

Reply:
quickly edited the code, sorry. The main idea was "you can reduce the size here by getting ride of `window.hlx` 2x"